lib/ProductOpener/Ecoscore.pm

```diff
@@ -1466,7 +1469,68 @@ sub localize_ecoscore ($$) {
 		}
 		
 	}		
+}
+
+
+
+=head2 ecoscore_extended_data_expected_error (  $product_ref)
+
+Expected error of the Eco-Score extended data from the impact estimator,
+based on % of uncharacterized ingredients and standard deviation.
+
+=head3 Arguments
+
+=head4 Product reference $product_ref
+
+=head3 Return values
+
+The expected error as float.
+
+=cut
+
+sub ecoscore_extended_data_expected_error ($) {
 	
+	my $product_ref = shift;
+
+	# Parameters of the surface, as generated by 
+	# https://github.com/openfoodfacts/off-product-environmental-impact/blob/master/analysis/colab/OFF%20impact%20estimator.ipynb
+	my @p = [ 0.16537831,  0.2269159 ,  0.04220039, -0.01991893,  0.44583949,
+       -0.06321924, -0.37268731,  0.12465602, -0.09215003,  0.06000644];
+
+	my $stddev = $product_ref->{ecoscore_extended_data}{ef_single_score_log_stddev};
+	my $unchar = $product_ref->{ecoscore_extended_data}{mass_ratio_uncharacterized};
+
+	return $p[0] +
+		$p[1] * $unchar +
+		$p[2] * $stddev +
+		$p[3] * $unchar * $stddev +
+		$p[4] * $unchar * $unchar +
+		$p[5] * $stddev * $stddev +
+		$p[6] * $unchar * $unchar * $unchar +
+		$p[7] * $unchar * $unchar * $stddev +
+		$p[8] * $unchar * $stddev * $stddev +
+		$p[9] * $stddev * $stddev * $stddev;
+
+}
+
+sub is_ecoscore_extended_data_more_precise_than_agribalyse ($) {
+
+	my $product_ref = shift;
+
+	# Check that the product has both Agribalyse and Impact Estimator data
+
+	my $agribalyse_score = deep_get($product_ref, qw(agribalyse ef_agriculture));
+	my $estimated_score = deep_get($product_ref, qw(ecoscore_extended_data impact likeliest_impacts EF_single_score));
+
+	
+	if ((defined $agribalyse_score) and (defined $estimated_score)) {
+
+		return (ecoscore_extended_data_expected_error($product_ref) 
+			< abs((log($agribalyse_score) - log($estimated_score)) / log($estimated_score)));
+	}
```

Comment:
There again, I imagine you double checked with Martin ? (Sorry but as we did not find examples in our test data set, I think we'd better double check).
There are too much differences in naming with ipynb, I'm not sure…

Reply:
I could do a pair programming session with this if you wanted?